Nodejs/Product/TypeScriptSourceMapReader/Serialization/BinarySerializer.cs

```diff
@@ -0,0 +1,95 @@
+﻿//*********************************************************//
+//    Copyright (c) Microsoft. All rights reserved.
+//    
+//    Apache 2.0 License
+//    
+//    You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//    
+//    Unless required by applicable law or agreed to in writing, software 
+//    distributed under the License is distributed on an "AS IS" BASIS, 
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 
+//    implied. See the License for the specific language governing 
+//    permissions and limitations under the License.
+//
+//*********************************************************//
+
+using System;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.NodejsTools.TypeScriptSourceMapReader {
+    /// <summary>
+    /// Custom serializer to convert the different types to byte[]
+    /// The serialization and deserialization needs to be always in sync
+    /// </summary>
+    internal class BinarySerializer {
```

Comment:
This should probably be an `IDisposable` and be used in `using` statements in order to ensure that its members are properly disposed, even in exceptional flows.